### ISSUE TITLE
Crash Report

### ISSUE THREAD

**Issue by Oisin-JK** [qgis/QGIS#55446]
### What is the bug or the crash?

Creating a new report. Deleting static table layout item.

## Report Details

**Python Stack Trace**
```
Windows fatal exception: access violation

Current thread 0x000009d0 (most recent call first):
<no Python frame>
```

**Stack Trace**

```

QGraphicsItem::dragMoveEvent :
QGraphicsEffectSource::draw :
QgsLayoutEffect::draw :
QGraphicsScene::drawItems :
QGraphicsScene::drawItems :
QGraphicsView::paintEvent :
QgsLayoutView::paintEvent :
QWidget::event :
QFrame::event :
QGraphicsView::viewportEvent :
QCoreApplicationPrivate::sendThroughObjectEventFilters :
QApplicationPrivate::notify_helper :
QApplication::notify :
QgsApplication::notify :
QCoreApplication::notifyInternal2 :
QWidgetPrivate::drawWidget :
QWidgetPrivate::paintSiblingsRecursive :
QWidgetPrivate::drawWidget :
QWidgetPrivate::paintSiblingsRecursive :
QWidgetPrivate::drawWidget :
QWidgetPrivate::paintSiblingsRecursive :
QWidgetPrivate::drawWidget :
QWidgetPrivate::paintSiblingsRecursive :
QWidgetPrivate::drawWidget :
QWidgetPrivate::moveRect :
QWidget::event :
QApplicationPrivate::notify_helper :
QApplication::notify :
QgsApplication::notify :
QCoreApplication::notifyInternal2 :
QCoreApplicationPrivate::sendPostedEvents :
QGraphicsScene::~QGraphicsScene :
QObject::event :
QGraphicsScene::event :
QApplicationPrivate::notify_helper :
QApplication::notify :
QgsApplication::notify :
QCoreApplication::notifyInternal2 :
QCoreApplicationPrivate::sendPostedEvents :
qt_plugin_query_metadata :
QEventDispatcherWin32::processEvents :
qt_plugin_query_metadata :
QEventLoop::exec :
QCoreApplication::exec :
main :
BaseThreadInitThunk :
RtlUserThreadStart :
```

### Steps to reproduce the issue

Unpredicatable crash. Unable to reproduce reliably.

### Versions

**QGIS Info**
QGIS Version: 3.28.11-Firenze
QGIS code revision: b02458a196
Compiled against Qt: 5.15.3
Running against Qt: 5.15.3
Compiled against GDAL: 3.7.2
Running against GDAL: 3.7.2

**System Info**
CPU Type: x86_64
Kernel Type: winnt
Kernel Version: 10.0.19045

### Supported QGIS version

- [ ] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [ ] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

### Additional context

One of numewrous recent crashes when editing Layouts or Reports

**Comment by Oisin-JK** [comment]
And another crash

## User Feedback

Remving and renaming attributes of point, line and polygon layers.

## Report Details

**Python Stack Trace**
```
Windows fatal exception: access violation

Current thread 0x00001190 (most recent call first):
<no Python frame>
```


**Stack Trace**

```

QByteArray::QByteArray :
QgsProviderSublayerDetails::name :
QgsPolymorphicRelation::generateRelations :
QgsRelationManagerDialog::addPolymorphicRelation :
QgsRelationManagerDialog::setLayers :
QgsProjectProperties::QgsProjectProperties :
QgisApp::projectProperties :
QgisApp::identifyMapTool :
QObject::qt_static_metacall :
QAction::activate :
QMenu::actionGeometry :
QMenu::actionGeometry :
QMenu::mouseReleaseEvent :
QWidget::event :
QMenu::event :
QApplicationPrivate::notify_helper :
QApplication::notify :
QgsApplication::notify :
QCoreApplication::notifyInternal2 :
QApplicationPrivate::sendMouseEvent :
QSizePolicy::QSizePolicy :
QSizePolicy::QSizePolicy :
QApplicationPrivate::notify_helper :
QApplication::notify :
QgsApplication::notify :
QCoreApplication::notifyInternal2 :
QGuiApplicationPrivate::processMouseEvent :
QWindowSystemInterface::sendWindowSystemEvents :
QEventDispatcherWin32::processEvents :
qt_plugin_query_metadata :
QEventLoop::exec :
QCoreApplication::exec :
main :
BaseThreadInitThunk :
RtlUserThreadStart :
```



**QGIS Info**
QGIS Version: 3.28.11-Firenze
QGIS code revision: b02458a196
Compiled against Qt: 5.15.3
Running against Qt: 5.15.3
Compiled against GDAL: 3.7.2
Running against GDAL: 3.7.2



**System Info**
CPU Type: x86_64
Kernel Type: winnt
Kernel Version: 10.0.19045

**Comment by nyalldawson** [comment]
The first bug was fixed in 3.34 (and the changes are too intrusive to backport to the LTR release). You'll need to file a separate issue for your second crash, that's completely unrelated.
